docs/components/pagination.md

```diff
@@ -0,0 +1,133 @@
+# Pagination
+
+**KPagination** is a component used for pagination.
+
+<template>
+  <KPagination :totalCount="300"/>
+</template>
+
+## Props
+### Total count - required
+A total number of items inside the paginated data source.
+
+```vue
+<KPagination :totalCount="50"/>
+```
+
+### Page sizes
+A list of page sizes that the user can choose from.
+
+Prop is an Array of numbers. The default page sizes are: [15, 25, 50, 75, 100].
+
+You can provide custom page sizes. The first one in the array will be the initial page size.
+
+<KPagination :totalCount="100" :pageSizes="[10, 20, 30, 40]"/>
+
+```vue
+<KPagination :totalCount="100" :pageSizes="[10, 20, 30, 40]"/>
+```
+
+### Neighbours
+A number that sets the neighbouring pages visible in a single view. For bigger sets of data we want user to see more pages to go through the pagination faster.
+
+<template>
+  <KPagination :totalCount="1000" :neighbours="2"/>
+</template>
+
+```vue
+<KPagination :totalCount="1000" :pageSize="15" neighbours="2"/>
+```
+
+## Usage
+
+<Komponent :data="{ names: ['Alice', 'Bob', 'Charlie', 'Derek', 'Ellie', 'Frank', 'George', 'Helen', 'Ingrid', 'Herman'], visibleNames: ['Alice', 'Bob', 'Charlie'], page: 1}" v-slot="{ data }">
+  <div>
+    <KCard title="Cool names list" class="mb-4">
+      <template #body>
+        <div v-for="name in data.visibleNames">
+        {{name}}
+        </div>
+      </template>
+    </KCard>
+
+  <KPagination 
+    :items="data.names"
+    :totalCount="data.names.length" 
+    :pageSizes="[3, 4, 5]" 
+    @pageChanged="({visibleItems}) => data.visibleNames = visibleItems"/>
+  </div>
+</Komponent>
+
+```vue
+<template>
+  <div>
+    <KCard title="Cool names list">
+      <template #body>
+        <div v-for="name in data.visibleNames">
+        {{name}}
+        </div>
+      </template>
+    </KCard>
+
+    <KPagination 
+      :items="data.names"
+      :totalCount="data.names.length" 
+      :pageSizes="[3, 4, 5]" 
+      @pageChanged="({visibleItems}) => data.visibleNames = visibleItems"/>
+  </div>
+</template>
+
+<script>
+export default {
+  data () {
+    const names = [
+      'Alice', 'Bob', 'Charlie', 
+      'Derek', 'Ellie', 'Frank', 
+      'George', 'Helen', 'Ingrid'
+    ]
+    return {
+      names
+      visibleNames: names.slice(0,3);
+      page: 1
+    }
+  }
+}
+</script>
+```
+
+## Theming
```

Comment:
Can we add an `Events` section that documents what events are emitted and what props are passed on those events?

Reply:
Done